src/message/contentTypes/textMessage.js

```diff
@@ -10,7 +10,13 @@ export default class TextMessage extends Message {
 
     setText(text) {
         validate.stringLength(text, null, 640, 'text', 'TextMessage.setText');
-        this.text = text;
+        this.state.text = text;
+        return this;
+    }
+
+    withTag(tag) {
+        validate.oneOf(tag, ['ISSUE_RESOLUTION'], 'tag', 'TextMessage.withTag');
+        this.tag = tag;
```

Comment:
Shouldn't this go on this.state.tag?  Seem to be updating that pattern everywhere